src/tests/utils/test_windows.rs

```diff
@@ -554,28 +563,331 @@ mod command_line_test {
+        }
+    }
+
+    mod window_handle_tests {
+        use windows::Win32::{
+            System::Threading::PROCESS_QUERY_INFORMATION,
+            UI::WindowsAndMessaging::GetWindowThreadProcessId,
         };
-        assert!(windows_api.get_exit_code(process_info.hProcess).unwrap() == STILL_ACTIVE.0 as u32);
-        unsafe { TerminateProcess(process_info.hProcess, 0) }.expect("Failed to terminate process");
-        assert!(windows_api.get_exit_code(process_info.hProcess).unwrap() == 0);
+
+        use crate::utils::windows::{DefaultWindowsApi, WindowsApi};
+
+        struct TestProcessGuard {
+            process_info: windows::Win32::System::Threading::PROCESS_INFORMATION,
+        }
+
+        impl Drop for TestProcessGuard {
+            fn drop(&mut self) {
+                match unsafe {
+                    windows::Win32::System::Threading::TerminateProcess(
+                        self.process_info.hProcess,
+                        0,
+                    )
+                } {
+                    Ok(_) => {}
+                    Err(err) => {
+                        if err.code().0 == 0x80070005u32 as i32 {
+                            eprintln!("Access denied when terminating process, that's okay");
+                        } else {
+                            panic!("Failed to terminate process: {}", err);
+                        }
+                    }
+                }
+            }
+        }
+
+        fn get_test_window_handle(
+            windows_api: &DefaultWindowsApi,
+        ) -> (windows::Win32::Foundation::HWND, TestProcessGuard) {
+            let application = r"C:\Windows\System32\timeout.exe";
+            let args = vec!["30".to_string()];
+            let process_info = windows_api
+                .create_process_with_args(application, args)
+                .expect("Failed to create process");
+            let hwnd = windows_api.get_window_handle_for_process(process_info.dwProcessId);
+            return (hwnd, TestProcessGuard { process_info });
+        }
+
+        mod console_title_test {
+            use crate::utils::windows::{
+                get_console_title,
+                test_mod::default_windows_api_tests::window_handle_tests::get_test_window_handle,
+                DefaultWindowsApi, WindowsApi,
+            };
+
+            /// Tests setting and getting console title.
+            /// Validates that the console title is correctly set and retrieved.
+            #[test]
+            fn test_set_and_get_console_title() {
+                let windows_api = DefaultWindowsApi;
+                let (handle, _guard) = get_test_window_handle(&windows_api);
+                let test_title = "Test Console Title";
+
+                windows_api
+                    .set_console_title(test_title, Some(handle))
+                    .expect("Failed to set console title");
+
+                let set_title = get_console_title(&windows_api);
```

Comment:
Why is this a dedicated function? Can we get rid of it?
We need to pass the handle in, thats why the test is failing